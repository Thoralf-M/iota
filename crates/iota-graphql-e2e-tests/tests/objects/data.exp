processed 5 tasks

init:
A: object(0,0)

task 1, lines 7-40:
//# publish
created: object(1,0)
mutated: object(0,1)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 7615200,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, lines 42-44:
//# programmable --inputs @A
//> 0: P0::m::foo();
//> TransferObjects([Result(0)], Input(0))
created: object(2,0)
mutated: object(0,1)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 2895600,  storage_rebate: 980400, non_refundable_storage_fee: 0

task 3, line 46:
//# create-checkpoint
Checkpoint created: 1

task 4, lines 48-69:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "asMoveObject": {
                      "contents": {
                        "type": {
                          "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
                        },
                        "data": {
                          "Struct": [
                            {
                              "name": "id",
                              "value": {
                                "UID": [
                                  144,
                                  13,
                                  164,
                                  144,
                                  250,
                                  243,
                                  84,
                                  252,
                                  106,
                                  167,
                                  63,
                                  227,
                                  132,
                                  230,
                                  67,
                                  229,
                                  236,
                                  255,
                                  205,
                                  61,
                                  4,
                                  205,
                                  189,
                                  44,
                                  92,
                                  229,
                                  190,
                                  37,
                                  0,
                                  39,
                                  41,
                                  14
                                ]
                              }
                            },
                            {
                              "name": "balance",
                              "value": {
                                "Struct": [
                                  {
                                    "name": "value",
                                    "value": {
                                      "Number": "299999988469600"
                                    }
                                  }
                                ]
                              }
                            }
                          ]
                        },
                        "json": {
                          "id": "0x900da490faf354fc6aa73fe384e643e5ecffcd3d04cdbd2c5ce5be250027290e",
                          "balance": {
                            "value": "299999988469600"
                          }
                        }
                      }
                    }
                  }
                },
                {
                  "outputState": {
                    "asMoveObject": {
                      "contents": {
                        "type": {
                          "repr": "0x2a5ff6568d23a1a54e88d86485e3bd4158f613493b350464d47bba94d0b953ac::m::Foo"
                        },
                        "data": {
                          "Struct": [
                            {
                              "name": "id",
                              "value": {
                                "UID": [
                                  197,
                                  234,
                                  223,
                                  234,
                                  140,
                                  89,
                                  124,
                                  216,
                                  180,
                                  58,
                                  98,
                                  86,
                                  219,
                                  98,
                                  4,
                                  255,
                                  86,
                                  115,
                                  131,
                                  58,
                                  221,
                                  48,
                                  161,
                                  176,
                                  117,
                                  194,
                                  191,
                                  10,
                                  35,
                                  172,
                                  85,
                                  122
                                ]
                              }
                            },
                            {
                              "name": "f0",
                              "value": {
                                "ID": [
                                  197,
                                  234,
                                  223,
                                  234,
                                  140,
                                  89,
                                  124,
                                  216,
                                  180,
                                  58,
                                  98,
                                  86,
                                  219,
                                  98,
                                  4,
                                  255,
                                  86,
                                  115,
                                  131,
                                  58,
                                  221,
                                  48,
                                  161,
                                  176,
                                  117,
                                  194,
                                  191,
                                  10,
                                  35,
                                  172,
                                  85,
                                  122
                                ]
                              }
                            },
                            {
                              "name": "f1",
                              "value": {
                                "Bool": true
                              }
                            },
                            {
                              "name": "f2",
                              "value": {
                                "Number": "42"
                              }
                            },
                            {
                              "name": "f3",
                              "value": {
                                "Number": "43"
                              }
                            },
                            {
                              "name": "f4",
                              "value": {
                                "String": "hello"
                              }
                            },
                            {
                              "name": "f5",
                              "value": {
                                "String": "world"
                              }
                            },
                            {
                              "name": "f6",
                              "value": {
                                "Vector": [
                                  {
                                    "Address": [
                                      197,
                                      234,
                                      223,
                                      234,
                                      140,
                                      89,
                                      124,
                                      216,
                                      180,
                                      58,
                                      98,
                                      86,
                                      219,
                                      98,
                                      4,
                                      255,
                                      86,
                                      115,
                                      131,
                                      58,
                                      221,
                                      48,
                                      161,
                                      176,
                                      117,
                                      194,
                                      191,
                                      10,
                                      35,
                                      172,
                                      85,
                                      122
                                    ]
                                  }
                                ]
                              }
                            },
                            {
                              "name": "f7",
                              "value": {
                                "Option": {
                                  "Number": "44"
                                }
                              }
                            }
                          ]
                        },
                        "json": {
                          "id": "0xc5eadfea8c597cd8b43a6256db6204ff5673833add30a1b075c2bf0a23ac557a",
                          "f0": "0xc5eadfea8c597cd8b43a6256db6204ff5673833add30a1b075c2bf0a23ac557a",
                          "f1": true,
                          "f2": 42,
                          "f3": "43",
                          "f4": "hello",
                          "f5": "world",
                          "f6": [
                            "0xc5eadfea8c597cd8b43a6256db6204ff5673833add30a1b075c2bf0a23ac557a"
                          ],
                          "f7": 44
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
