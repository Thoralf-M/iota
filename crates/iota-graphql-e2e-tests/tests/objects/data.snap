---
source: external-crates/move/crates/move-transactional-test-runner/src/framework.rs
---
processed 5 tasks

init:
A: object(0,0)

task 1, lines 7-40:
//# publish
created: object(1,0)
mutated: object(0,1)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 7615200,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, lines 42-44:
//# programmable --inputs @A
//> 0: P0::m::foo();
//> TransferObjects([Result(0)], Input(0))
created: object(2,0)
mutated: object(0,1)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 2895600,  storage_rebate: 980400, non_refundable_storage_fee: 0

task 3, line 46:
//# create-checkpoint
Checkpoint created: 1

task 4, lines 48-69:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "asMoveObject": {
                      "contents": {
                        "type": {
                          "repr": "0x0000000000000000000000000000000000000000000000000000000000000002::coin::Coin<0x0000000000000000000000000000000000000000000000000000000000000002::iota::IOTA>"
                        },
                        "data": {
                          "Struct": [
                            {
                              "name": "id",
                              "value": {
                                "UID": [
                                  116,
                                  24,
                                  70,
                                  16,
                                  235,
                                  27,
                                  229,
                                  56,
                                  50,
                                  217,
                                  59,
                                  92,
                                  3,
                                  91,
                                  107,
                                  47,
                                  195,
                                  237,
                                  24,
                                  142,
                                  185,
                                  206,
                                  191,
                                  182,
                                  166,
                                  109,
                                  63,
                                  96,
                                  171,
                                  94,
                                  76,
                                  214
                                ]
                              }
                            },
                            {
                              "name": "balance",
                              "value": {
                                "Struct": [
                                  {
                                    "name": "value",
                                    "value": {
                                      "Number": "299999988469600"
                                    }
                                  }
                                ]
                              }
                            }
                          ]
                        },
                        "json": {
                          "id": "0x74184610eb1be53832d93b5c035b6b2fc3ed188eb9cebfb6a66d3f60ab5e4cd6",
                          "balance": {
                            "value": "299999988469600"
                          }
                        }
                      }
                    }
                  }
                },
                {
                  "outputState": {
                    "asMoveObject": {
                      "contents": {
                        "type": {
                          "repr": "0xc8d463f11158f26ceba794706b7b717132b6f5a39a32e740d74da188d4cabd42::m::Foo"
                        },
                        "data": {
                          "Struct": [
                            {
                              "name": "id",
                              "value": {
                                "UID": [
                                  252,
                                  18,
                                  126,
                                  214,
                                  120,
                                  237,
                                  255,
                                  110,
                                  123,
                                  175,
                                  126,
                                  84,
                                  72,
                                  127,
                                  127,
                                  99,
                                  165,
                                  123,
                                  247,
                                  163,
                                  124,
                                  86,
                                  115,
                                  177,
                                  242,
                                  19,
                                  35,
                                  58,
                                  152,
                                  243,
                                  52,
                                  34
                                ]
                              }
                            },
                            {
                              "name": "f0",
                              "value": {
                                "ID": [
                                  252,
                                  18,
                                  126,
                                  214,
                                  120,
                                  237,
                                  255,
                                  110,
                                  123,
                                  175,
                                  126,
                                  84,
                                  72,
                                  127,
                                  127,
                                  99,
                                  165,
                                  123,
                                  247,
                                  163,
                                  124,
                                  86,
                                  115,
                                  177,
                                  242,
                                  19,
                                  35,
                                  58,
                                  152,
                                  243,
                                  52,
                                  34
                                ]
                              }
                            },
                            {
                              "name": "f1",
                              "value": {
                                "Bool": true
                              }
                            },
                            {
                              "name": "f2",
                              "value": {
                                "Number": "42"
                              }
                            },
                            {
                              "name": "f3",
                              "value": {
                                "Number": "43"
                              }
                            },
                            {
                              "name": "f4",
                              "value": {
                                "String": "hello"
                              }
                            },
                            {
                              "name": "f5",
                              "value": {
                                "String": "world"
                              }
                            },
                            {
                              "name": "f6",
                              "value": {
                                "Vector": [
                                  {
                                    "Address": [
                                      252,
                                      18,
                                      126,
                                      214,
                                      120,
                                      237,
                                      255,
                                      110,
                                      123,
                                      175,
                                      126,
                                      84,
                                      72,
                                      127,
                                      127,
                                      99,
                                      165,
                                      123,
                                      247,
                                      163,
                                      124,
                                      86,
                                      115,
                                      177,
                                      242,
                                      19,
                                      35,
                                      58,
                                      152,
                                      243,
                                      52,
                                      34
                                    ]
                                  }
                                ]
                              }
                            },
                            {
                              "name": "f7",
                              "value": {
                                "Option": {
                                  "Number": "44"
                                }
                              }
                            }
                          ]
                        },
                        "json": {
                          "id": "0xfc127ed678edff6e7baf7e54487f7f63a57bf7a37c5673b1f213233a98f33422",
                          "f0": "0xfc127ed678edff6e7baf7e54487f7f63a57bf7a37c5673b1f213233a98f33422",
                          "f1": true,
                          "f2": 42,
                          "f3": "43",
                          "f4": "hello",
                          "f5": "world",
                          "f6": [
                            "0xfc127ed678edff6e7baf7e54487f7f63a57bf7a37c5673b1f213233a98f33422"
                          ],
                          "f7": 44
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
